doc/source/building/windows.rst

```diff
@@ -8,155 +8,268 @@ Building From Source on Windows
+
+    # The following two lines clean up in case we make a mistake and need
+    # to run the script again
+    git clean -fxd
+    rm -rf $OPENBLAS_ROOT/$BUILD_BITS
+
+    # Set architecture flags
+    march="x86-64"
+    extra="-fno-asynchronous-unwind-tables"
+    vc_arch="X64"
+    cflags="-O2 -march=$march -mtune=generic $extra"
+    fflags="$cflags -frecursive -ffpe-summary=invalid,zero"
+
+    # Build name for output library from gcc version and OpenBLAS commit.
+    GCC_TAG="gcc_$(gcc -dumpversion | tr .- _)"
+    OPENBLAS_VERSION=$(git describe --tags)
+    # Build OpenBLAS
+    # Variable used in creating output libraries
+    export LIBNAMESUFFIX=${OPENBLAS_VERSION}-${GCC_TAG}
+    make BINARY=$BUILD_BITS DYNAMIC_ARCH=1 USE_THREAD=1 USE_OPENMP=0 \ 
+        NO_WARMUP=1 BUILD_LAPACK_DEPRECATED=1 \ 
+        COMMON_OPT="$cflags" FCOMMON_OPT="$fflags"
+    make install PREFIX=$OPENBLAS_ROOT/$BUILD_BITS
+
+This is the automation script that will make sure the right variables are used
+in the right place. Linux users are very familiar to such scripts but for
+Windows users it might be a bit awkward. You can think of these as ``.bat``
+files. You can change the variables to your situation. After you've created this
+file and you are one directory up the OpenBLAS repo of that, start the OpenBLAS
+build with:
+
+.. code:: shell
+
+    ./build_openblas.sh
 
-After the :code:`build_openblas.sh` script has completed (probably with an error), there
-should be an :code:`openblas.a` file somewhere on your system. If :code:`OPENBLAS_ROOT` was
-set to :code:`C:\\opt`, then you might see a line like this in the MSYS2 terminal:
+Building OpenBLAS is challenging and time-consuming. The build may fail with an
+error after a few hours but may also fail silently and produce an incorrect
+binary. Please, if you have any issues, `report them`_ so that we can save the
+next person's time.
+
+One of the known issues is the following; if you, by any chance, receive the
+following error
+
+.. code::shell
+
+    <command-line>:0:4: error: expected identifier or '(' before numeric constant
+    
+that means you have some header file definition clash and you have to downgrade
+certain items. This is not related to SciPy but let's attempt to provide a
+solution. See this
+`OpenBLASwiki <https://github.com/xianyi/OpenBLAS/wiki/How-to-use-OpenBLAS-in-Microsoft-Visual-Studio#build-openblas-on-windows-os>`__
+page to read on which packages to downgrade and how to do it.
+Basically, it involves downloading three files. Then in the MSYS terminal chage
+the directory to the place where you downloaded the files and run the commands
+given in the wiki link. Then come back to the script directory where
+`./build_openblas.sh` lives and try again. This should be sufficient for you to
+build OpenBLAS.
+
+While you're waiting on OpenBLAS to finish building, go ahead and install
+`build tools`_ from Microsoft, since these take a while to install and you'll 
```

Comment:
"Build Tools for Visual Studio 2019"?

Reply:
Yes indeed. The next question is what to choose. I'll add some as far as I remember

Reply:
I would write out the full name of the product to download as it appears at the linked site.